src/main/java/ch/njol/skript/classes/data/BukkitClasses.java

```diff
@@ -1548,6 +1553,48 @@ public String toVariableNameString(EnchantmentOffer eo) {
 				.name("Transform Reason")
 				.description("Represents a transform reason of an <a href='events.html#entity transform'>entity transform event</a>.")
 				.since("2.8.0"));
+
+		if (Skript.classExists("org.bukkit.damage.DamageType")) {
+			Classes.registerClass(new ClassInfo<>(DamageSource.class, "damagesource")
+					.user("damage ?sources?")
+					.name("Damage Source")
+					.description("The damage source in an entity damage event.")
+					.since("INSERT VERSION"));
+
+			Classes.registerClass(new ClassInfo<>(DamageType.class, "damagetype")
+					.user("damage ?types?")
+					.name("Damage Type")
+					.description("The damage type of a damage source.")
+					.since("INSERT VERSION")
+					.parser(new Parser<DamageType>() {
+						@Override
+						@Nullable
+						public DamageType parse(String input, ParseContext context) {
+							if (input.contains(":")) {
+								String[] split = input.split(Pattern.quote(":"));
+								try {
+									return Registry.DAMAGE_TYPE.get(new NamespacedKey(split[0], split[1]));
+								} catch (IllegalArgumentException e) {}
+							}
+							try {
+								return Registry.DAMAGE_TYPE.get(NamespacedKey.minecraft(input));
+							} catch (IllegalArgumentException e) {}
+
+							return null;
+						}
+
+						@Override
+						public String toString(DamageType type, int flags) {
+							return type.getKey().getNamespace();
+						}
+
+						@Override
+						public String toVariableNameString(DamageType type) {
+							return "damage type:" + type.getKey().getNamespace();
```

Comment:
`.getNamespace()` will return "minecraft"

```suggestion
						public String toString(DamageType type, int flags) {
							return type.getKey().geyKey();
						}

						@Override
						public String toVariableNameString(DamageType type) {
							return "damage type:" + type.getKey().geyKey();
```

Reply:
Yes, that's the point because data packs can register their own key and value